File: minwook/5주차/약수의_개수와_덧셈.java
class Solution {
    public int solution(int left, int right) {
        int answer = 0;
        for(int n=left; n<=right; n++){
            if(measureEvenOdd(n) % 2 == 0) // 짝수, 홀수 구분
                answer += n;
            else
                answer -= n;
        }
        return answer;
    }
    
    public int measureEvenOdd(int num){ // 약수 구하기
        int evenOdd = 0;
        for(int i=1; i<=num; i++){
            if(num % i == 0)
                evenOdd++;
        }
        return evenOdd;
    }
}

File: yeongseong/5week/약수의 개수와 덧셈.js
function solution(left, right) {
    var answer = 0;
    for (let i = left; i <= right; i++) {
      let count = 0; 
      for (let j = 1; j <= i; j++) {
        if (i % j === 0) { // 현재 숫자를 j로 나누어 약수인지 확인
          count += 1; // 약수이면 count 증가
        }
      }
      if (count % 2 === 0) { // 약수 개수가 짝수인 경우
        answer += i; // 현재 숫자를 더함
      } else { // 약수 개수가 홀수인 경우
        answer -= i; // 현재 숫자를 뺌
      }
    }
    return answer; 
  }
  